From ffabc4d65c54ea72a7318251be44120c9e87c542 Mon Sep 17 00:00:00 2001
From: Carlo Bramini <30959007+carlo-bramini@users.noreply.github.com>
Date: Tue, 30 Jun 2020 09:37:54 +0200
Subject: [PATCH 1546/1936] Fix error in ZLIB detection

I'm getting this message on screen when CMake configures the project:

  The package name passed to `find_package_handle_standard_args` ;(ZLIB) does
  not match the name of the calling package ;(zlib).  This can lead to
  problems in calling ;code that expects `find_package` result variables
  (e.g., `_FOUND`) ;to follow a certain pattern.
Call Stack (most recent call first):
  /home/carlo/inst_cmake/share/cmake-3.17.3/Modules/FindPackageHandleStandardArgs.cmake:272 (message)
  cmake/Findzlib.cmake:39 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/CMakeLists.txt:13 (find_package)

This happens because the target must be "ZLIB", with capital letters, and not "zlib":

https://cmake.org/cmake/help/git-master/module/FindZLIB.html
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f6c7af1d..804831f3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,7 +10,7 @@ include(CheckCXXCompilerFlag)
 if(NOT (${CMAKE_SYSTEM_NAME} STREQUAL "Windows"))
     find_package(PkgConfig)
 endif()
-find_package(zlib REQUIRED)
+find_package(ZLIB REQUIRED)
 find_package(X11)
 find_package(Threads   REQUIRED)
 find_package(OSS)
-- 
2.47.0

